### DISCUSSION TITLE
Section 4 Solution to "Slither: command not found"

### DISCUSSION THREAD

**Discussion by kwamentw** [Cyfrin/security-and-auditing-full-course-s23#12] [Ideas]
Linux platforms are sometimes confused on what python package to run slither with so just specify like this;

```bash
python3 -m slither
```
when in the project directory too you can run

```bash
python3 -m slither .
```

**Comment by iftikharuddin** [comment]
Make sense, I am thinking if there is a way to just run slither on a single Contract? e.g MyToken.sol

**Reply by drained69** [reply]
to run Slither on a full folder on linux, you can use ```python3 -m slither .``` while for a src like MyToken.sol you can use ``` python3 -m slither ./src/MyToken.sol```

**Comment by cromewar** [comment]
Thanks @kwamentw

**Comment by MartinPetrov12** [comment]
Hey, Linux user here! 

Whenever i try to run:
`python3 -m slither .` 

i get the same error: 
`ImportError: cannot import name 'Slither' from 'slither' (/home/martin/.local/lib/python3.10/site-packages/slither/__init__.py)`

Any idea why that could be happening?

**Reply by kwamentw** [reply]
Try the command again but without the `.` at the end and let's see will be waiting for your reply

**Reply by drained69** [reply]
Hmmm, his issue seems to be his device not been able to identify slither. He could try reinstalling it maybe.

**Reply by MartinPetrov12** [reply]
Sorry for not responding earlier guys, my bad. I tried without the `.`, but still got the same error. Also tried reinstalling it, no effect either.

**Reply by drained69** [reply]
Sorry I know issues like this can be stressful but then honestly seems like an installation/storageissue and hard to tell the cause here from my part. How did you go about your installation?

**Reply by MartinPetrov12** [reply]
Somehow, after uninstalling it again, this time it worked. I think the issue was that i had installed both `slither` and `slither-analyzer`. After removing both of them and reinstalling `slither-analyzer` everything fell into place. Thanks to everyone for the help!

**Comment by giriraj-kulkarni** [comment]
hey guyz i get this error


ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (ckzg, cytoolz, lru-dict)

**Reply by EngrPips** [reply]
This is the first time I am seeing this error. Have you checked the official Slither https://github.com/crytic/slither Github to see if anyone has reported similar issue before.
